### ISSUE TITLE
Error using YOLO-NAS with Ultralytics

### ISSUE THREAD

**Issue by FaresElmenshawi** [Deci-AI/super-gradients#1640]
### 🐛 Describe the bug

I am trying to use YOLO-Nas with ultralytics and the model isn't being automatically downloaded and I'm getting the error below when trying to run inference. I've asked about this issue on their discord and was told that the issue came from `super_gradients`

# Sample code to reproduce the problem
```python
from ultralytics import NAS

model = NAS('Models/yolo_nas_s.pt')

results = model.predict('images/img.jpg')
```

# The error message you got, with the full traceback.

```
The console stream is logged into C:\Users\Fares\sg_logs\console.log
[2023-11-14 17:54:41] INFO - crash_tips_setup.py - Crash tips is enabled. You can set your environment variable to CRASH_HANDLER=FALSE to disable it
[2023-11-14 17:54:42] WARNING - __init__.py - Failed to import pytorch_quantization
[2023-11-14 17:54:42,065] torch.distributed.elastic.multiprocessing.redirects: [WARNING] NOTE: Redirects are currently not supported in Windows or MacOs.
[2023-11-14 17:54:44] WARNING - calibrator.py - Failed to import pytorch_quantization
[2023-11-14 17:54:44] WARNING - export.py - Failed to import pytorch_quantization
[2023-11-14 17:54:44] WARNING - selective_quantization_utils.py - Failed to import pytorch_quantization
[2023-11-14 17:54:44] WARNING - env_sanity_check.py - Failed to verify operating system: Deci officially supports only Linux kernels. Some features may not work as expected.

Traceback (most recent call last):
  File "C:\Users\Fares\PycharmProjects\vision-analysis\debug.py", line 5, in <module>
    results = model.predict('images/img.jpg')
  File "C:\Users\Fares\AppData\Local\Programs\Python\Python310\lib\site-packages\ultralytics\engine\model.py", line 239, in predict
    return self.predictor.predict_cli(source=source) if is_cli else self.predictor(source=source, stream=stream)
  File "C:\Users\Fares\AppData\Local\Programs\Python\Python310\lib\site-packages\ultralytics\engine\predictor.py", line 198, in __call__
    return list(self.stream_inference(source, model, *args, **kwargs))  # merge list of Result into one
  File "C:\Users\Fares\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\utils\_contextlib.py", line 35, in generator_context
    response = gen.send(None)
  File "C:\Users\Fares\AppData\Local\Programs\Python\Python310\lib\site-packages\ultralytics\engine\predictor.py", line 246, in stream_inference
    self.model.warmup(imgsz=(1 if self.model.pt or self.model.triton else self.dataset.bs, 3, *self.imgsz))
  File "C:\Users\Fares\AppData\Local\Programs\Python\Python310\lib\site-packages\ultralytics\nn\autobackend.py", line 480, in warmup
    self.forward(im)  # warmup
  File "C:\Users\Fares\AppData\Local\Programs\Python\Python310\lib\site-packages\ultralytics\nn\autobackend.py", line 347, in forward
    y = self.model(im, augment=augment, visualize=visualize) if augment or visualize else self.model(im)
  File "C:\Users\Fares\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "C:\Users\Fares\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "C:\Users\Fares\AppData\Local\Programs\Python\Python310\lib\site-packages\super_gradients\training\models\detection_models\customizable_detector.py", line 88, in forward
    x = self.backbone(x)
  File "C:\Users\Fares\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "C:\Users\Fares\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "C:\Users\Fares\AppData\Local\Programs\Python\Python310\lib\site-packages\super_gradients\modules\detection_modules.py", line 83, in forward
    x = getattr(self, layer)(x)
  File "C:\Users\Fares\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "C:\Users\Fares\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "C:\Users\Fares\AppData\Local\Programs\Python\Python310\lib\site-packages\super_gradients\training\models\detection_models\yolo_nas\yolo_stages.py", line 171, in forward
    return self.conv(x)
  File "C:\Users\Fares\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "C:\Users\Fares\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "C:\Users\Fares\AppData\Local\Programs\Python\Python310\lib\site-packages\super_gradients\modules\qarepvgg_block.py", line 189, in forward
    x_3x3 = self.branch_3x3(inputs)
  File "C:\Users\Fares\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "C:\Users\Fares\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "C:\Users\Fares\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\nn\modules\container.py", line 215, in forward
    input = module(input)
  File "C:\Users\Fares\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "C:\Users\Fares\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "C:\Users\Fares\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\nn\modules\batchnorm.py", line 171, in forward
    return F.batch_norm(
  File "C:\Users\Fares\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\nn\functional.py", line 2478, in batch_norm
    return torch.batch_norm(
RuntimeError: Inference tensors do not track version counter.```

### Versions

Collecting environment information...
PyTorch version: 2.1.0+cu118
Is debug build: False
CUDA used to build PyTorch: 11.8
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 10 Home
GCC version: Could not collect
Clang version: Could not collect
CMake version: Could not collect
Libc version: N/A

Python version: 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.19045-SP0
Is CUDA available: True
CUDA runtime version: 11.8.89
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce GTX 1050
Nvidia driver version: 546.01
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:

Revision=

Versions of relevant libraries:
[pip3] facenet-pytorch==2.5.3
[pip3] numpy==1.23.0
[pip3] onnx==1.13.0
[pip3] onnx-simplifier==0.4.35
[pip3] onnxruntime==1.13.1
[pip3] pytorchvideo==0.1.5
[pip3] torch==2.1.0+cu118
[pip3] torchaudio==2.1.0+cu118
[pip3] torchmetrics==0.8.0
[pip3] torchvision==0.16.0+cu118
[conda] Could not collect

**Comment by Burhan-Q** [comment]
This has been reported in the Ultralytics repo as well https://github.com/ultralytics/ultralytics/issues/5345 I have not tried with `super-gradients=3.1.2` as the user in the issue recommends.

**Comment by Phyrokar** [comment]
> With super-gradients 3.4.1 and the .perdict function you run into a similar error:
> 
> ```
> Traceback (most recent call last):
>   File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
>     self.run()
>   File "/usr/lib/python3.8/threading.py", line 870, in run
>     self._target(*self._args, **self._kwargs)
>   File "/usr/local/lib/python3.8/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
>     return func(*args, **kwargs)
>   File "/home/phyro/catkin_ws/src/Family_detector/detect_V1_1.py", line 362, in inference2
>     prediction = net.predict("/home/phyro/catkin_ws/Pic_2.png")
>   File "/usr/local/lib/python3.8/dist-packages/super_gradients/training/models/detection_models/customizable_detector.py", line 202, in predict
>     return pipeline(images, batch_size=batch_size)  # type: ignore
>   File "/usr/local/lib/python3.8/dist-packages/super_gradients/training/pipelines/pipelines.py", line 111, in __call__
>     return self.predict_images(inputs, batch_size)
>   File "/usr/local/lib/python3.8/dist-packages/super_gradients/training/pipelines/pipelines.py", line 127, in predict_images
>     return self._combine_image_prediction_to_images(result_generator, n_images=len(images))
>   File "/usr/local/lib/python3.8/dist-packages/super_gradients/training/pipelines/pipelines.py", line 311, in _combine_image_prediction_to_images
>     images_predictions = [next(iter(images_predictions))]
>   File "/usr/local/lib/python3.8/dist-packages/super_gradients/training/pipelines/pipelines.py", line 165, in _generate_prediction_result
>     yield from self._generate_prediction_result_single_batch(batch_images)
>   File "/usr/local/lib/python3.8/dist-packages/super_gradients/training/pipelines/pipelines.py", line 187, in _generate_prediction_result_single_batch
>     preprocessed_image, processing_metadata = self.image_processor.preprocess_image(image=image.copy())
>   File "/usr/local/lib/python3.8/dist-packages/super_gradients/training/processing/processing.py", line 99, in preprocess_image
>     processed_image, metadata = processing.preprocess_image(image=processed_image)
>   File "/usr/local/lib/python3.8/dist-packages/super_gradients/training/processing/processing.py", line 254, in preprocess_image
>     processed_image = _pad_image(image=image, padding_coordinates=padding_coordinates, pad_value=self.pad_value)
>   File "/usr/local/lib/python3.8/dist-packages/super_gradients/training/transforms/utils.py", line 151, in _pad_image
>     return np.pad(image, pad_width=padding_values, mode="constant", constant_values=constant_values)
>   File "<__array_function__ internals>", line 200, in pad
>   File "/usr/local/lib/python3.8/dist-packages/numpy/lib/arraypad.py", line 801, in pad
>     values = _as_pairs(values, padded.ndim)
>   File "/usr/local/lib/python3.8/dist-packages/numpy/lib/arraypad.py", line 487, in _as_pairs
>     x = np.array(x)
> ```
> 
> ```
> with 3.2.1 you dont have any issues.
> ```

I figured out that 3.4.1 runs with numpy 1.23.0. One warning
`/usr/local/lib/python3.8/dist-packages/numpy/lib/arraypad.py:487: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.
  x = np.array(x)`
 is issued but the results seem to be correct.

**Comment by BloodAxe** [comment]
This warning was addressed in 3.5.0 release of SuperGradients 😎
